### PR TITLE
Adding code generation utilities used on the loading side of things

### PR DESCRIPTION
**Pull request by jiwalker-usgs**
There is likely more to add here when we get to the beans and such.

### CODE REVIEW

File: R/generate_code.R [hunk 1]
```diff
@@ -0,0 +1,82 @@
+#'@export
```
Comment (cschroed-usgs) [inline R/generate_code.R]: @jiwalker-usgs, thanks for checking this in. Could you please add some documentation indicating the purpose of this function? I'm thinking of the following audience for the docs : Developer Dude is trying to decide whether or not he can delete this code 3 years from now, when we may or may not still be using it.

**Comment by cschroed-usgs** [comment]
Looks like Appveyor is failing at git lfs.

File: R/generate_code.R (hunk 1 above)
Comment (jiwalker-usgs) [inline R/generate_code.R]: Sure, I'll add a better description of this with my next push.  I think I'm going to be doing this same pattern for some of the Java stuff (variable declaration and such).

Comment (cschroed-usgs) [inline R/generate_code.R]: sounds good!

**Comment by jiwalker-usgs** [comment]
@cschroed-usgs let me know if that makes enough sense for future developer dude.

**Comment by cschroed-usgs** [comment]
:+1:

**Pull request merged**